api/src/main/java/org/pmiops/workbench/billing/NotificationLogger.java

```diff
@@ -1,17 +1,50 @@
 package org.pmiops.workbench.billing;
 
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
 import org.pmiops.workbench.db.model.DbUser;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Service;
 
+// RW-3661: replace this file with an implementation which sends emails instead
+
 @Service
 public class NotificationLogger implements NotificationService {
 
   private static final Logger logger = LoggerFactory.getLogger(BillingProjectBufferService.class);
 
   @Override
-  public void alertUser(DbUser user, String msg) {
-    logger.info("\nTO: " + user.getEmail() + " MSG: " + msg);
+  public void alertUserFreeTierDollarThreshold(
+      DbUser user, double threshold, double currentUsage, double remainingBalance) {
+    final String logMsg =
+        String.format(
+            "User %s has passed the %.2f free tier dollar threshold.  Current total usage is $%.2f with remaining balance $%.2f",
+            user.getEmail(), threshold, currentUsage, remainingBalance);
+    logger.info(logMsg);
+  }
+
+  @Override
+  public void alertUserFreeTierTimeThreshold(
+      DbUser user, long daysRemaining, LocalDate expirationDate, double remainingDollarBalance) {
+
+    // TODO choose desired date format
+    final String dateStr =
+        DateTimeFormatter.ofPattern("MM/dd/yyyy")
+            .withZone(ZoneId.systemDefault())
+            .format(expirationDate);
+
+    final String logMsg =
+        String.format(
+            "User %s has %d days remaining until their expiration date of %s.  Current total usage is $%.2f",
+            user.getEmail(), daysRemaining, dateStr, remainingDollarBalance);
+    logger.info(logMsg);
+  }
+
+  @Override
+  public void alertUserFreeTierExpiration(DbUser user) {
+    final String logMsg = String.format("Free credits have expired for User %s", user.getEmail());
```

Comment:
Is it sufficient to just say that the credits have expired? I'm thinking the user would want to know if it was either the time or $ amount that was reached.

Reply:
The design does not have this.  In either case, the user action is the same - switch to BYOB or contact us.